dit-core/src/peer/store.rs

```diff
@@ -0,0 +1,42 @@
+use std::fs;
+use std::io;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::path::{Path, PathBuf};
+use crate::peer::DhtAddr;
+
+#[derive(Clone)]
+pub struct Store {
+    /// Location of the store.
+    dir: PathBuf,
+    blob_dir: PathBuf,
+}
+
+impl Store {
+    /// Open dit path, creating blob_dir for local added files
+    pub fn open(dir: impl AsRef<Path>) -> Self {
+        Self {
+            dir: dir.as_ref().to_owned(),
+            blob_dir: dir.as_ref().join("blobs"),
+        }
+    }
+
+    /// Copies a file into the store without notifying peers.
+    pub fn add_file(&mut self, path: impl AsRef<Path>) -> io::Result<()> {
+        let mut file = fs::File::open(&path)?;
+        let hash = DhtAddr::buffer_hash(file)?;
+        fs::copy(&path, &self.blob_dir.join(hash.to_string()))?;
+        Ok(())
+    }
+
+    /// Get a file from the blob store by using the hash value provided
+    pub fn cat(&self, file_hash: impl AsRef<str>) -> io::Result<File> {
```

Comment:
```suggestion
    pub fn open_file_by_hash(&self, file_hash: DhtAddr) -> io::Result<File> {
```

Deskriptiverer Name und der Aufrufer muss bei dem Hash nicht selber `.to_string()` aufrufen.